packages/contracts-bedrock/scripts/checks/check-interfaces.sh

```diff
@@ -0,0 +1,171 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+# This script checks for ABI consistency between interfaces and their corresponding contracts.
+# It compares the ABIs of interfaces (files starting with 'I') with their implementation contracts,
+# excluding constructors and certain predefined files. The script reports any differences found
+# and exits with an error if inconsistencies are detected.
+# NOTE: Script is fast enough but could be parallelized if necessary.
+
+# Parse flags
+no_diff=false
+if [[ "${1:-}" == "--no-diff" ]]; then
+    no_diff=true
+fi
+
+# Grab the directory of the contracts-bedrock package
+SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
+CONTRACTS_BASE=$(dirname "$(dirname "$SCRIPT_DIR")")
+
+# Define the files to exclude (glob patterns can be used)
+EXCLUDE_FILES=(
+    # External dependencies
+    "IMulticall3"
+    "IERC20"
+    "IERC721"
+    "IERC721Enumerable"
+    "IERC721Metadata"
+    "IERC721Upgradeable"
+    "IERC721Receiver"
+    "IERC1271"
+    "IERC165"
+    "IVotes"
+    "IBeacon"
+    "IProxyCreationCallback"
+    "IAutomate"
+    "IGelato1Balance"
+    "ERC721TokenReceiver"
+    "ERC1155TokenReceiver"
+    "ERC777TokensRecipient"
+    "Guard"
+    "GnosisSafeProxy"
+
+    # Foundry
+    "Common"
+    "Vm"
+    "VmSafe"
+
+    # EAS
+    "IEAS"
+    "ISchemaResolver"
+    "ISchemaRegistry"
+
+    # Kontrol
+    "KontrolCheatsBase"
+    "KontrolCheats"
+
+    # Error definition files
+    "CommonErrors"
+    "Errors"
+
+    # TODO: Interfaces that need to be fixed
+    "IPreimageOracle"
+    "IOptimismMintableERC721"
+    "IFaultDisputeGame"
+    "IOptimismSuperchainERC20"
+    "IInitializable"
+    "IOptimismMintableERC20"
+    "ILegacyMintableERC20"
+    "MintableAndBurnable"
+    "IDisputeGameFactory"
+    "IAddressManager"
+    "IWETH"
+    "IDelayedWETH"
+    "IAnchorStateRegistry"
+    "ICrossL2Inbox"
+    "IL2ToL2CrossDomainMessenger"
+    "KontrolInterfaces"
+)
+
+# Convert the exclude files array to a pipe-separated string
+EXCLUDE_PATTERN=$(IFS="|"; echo "${EXCLUDE_FILES[*]}")
+
+# Find all JSON files in the forge-artifacts folder
+JSON_FILES=$(find "$CONTRACTS_BASE/forge-artifacts" -type f -name "*.json" | grep -Ev "$EXCLUDE_PATTERN")
+
+# Initialize a flag to track if any issues are detected
+issues_detected=false
+
+# Keep track of any interfaces that have already been reported
+declare -A reported_interfaces
```

Comment:
On macOS this line errors for me, seems it's not available on macOS from [this old stackexchange answer](https://unix.stackexchange.com/a/428205).
```
./scripts/checks/check-interfaces.sh: line 90: declare: -A: invalid option
declare: usage: declare [-afFirtx] [-p] [name[=value] ...]
error: Recipe `interfaces-check-no-build` failed on line 68 with exit code 2
```

Reply:
Will approve if this works on mac

Reply:
Weird, this works for me on mac

Reply:
This is now working, so resolving